manager/integration/tests/test_settings.py

```diff
@@ -891,28 +891,30 @@ def test_setting_concurrent_rebuild_limit(client, core_api, volume_name):  # NOQ
 
     # While one volume is rebuilding, verify another volume is not
     # rebuilding and stuck in degrading state
-    rebuild_started = False
-    for i in range(RETRY_COUNTS):
+    v1_rebuild_started = False
+    v2_rebuild_started = False
+    for i in range(RETRY_COUNTS_LONG):
         volume1 = client.by_id_volume(volume1_name)
         volume2 = client.by_id_volume(volume2_name)
 
-        if volume1.rebuildStatus == [] and \
-                volume2.rebuildStatus == [] and \
-                rebuild_started is False:
-            continue
-        elif volume1.rebuildStatus == [] and \
-                volume2.rebuildStatus == [] and \
-                rebuild_started is True:
+        if volume1.rebuildStatus == [] and volume2.rebuildStatus == [] \
+                and v1_rebuild_started and v2_rebuild_started:
             break
         elif volume2.rebuildStatus == []:
-            assert volume1.rebuildStatus[0].state == "in_progress"
-            rebuild_started = True
+            if volume1.rebuildStatus and \
+                    volume1.rebuildStatus[0].state == "in_progress" and \
+                    len(volume1.replicas) == 3:
+                v1_rebuild_started = True
         elif volume1.rebuildStatus == []:
-            assert volume2.rebuildStatus[0].state == "in_progress"
-            rebuild_started = True
+            if volume2.rebuildStatus and \
+                    volume2.rebuildStatus[0].state == "in_progress" and \
+                    len(volume1.replicas) == 3:
+                v2_rebuild_started = True
 
-        time.sleep(RETRY_INTERVAL)
+        time.sleep(RETRY_INTERVAL_LONG)
 
+    assert v2_rebuild_started
+    assert v1_rebuild_started
```

Comment:
I am a little confused. What we want to test is volume 2 rebuilding wil run first, then after volume 1. However, here, we just check both are started to rebuild. It seems not what we want to test?

Reply:
Hi @innobead :
     After discussing with James previously, there were concerns about the order of rebuilding [between volume 1 and volume 2](https://github.com/longhorn/longhorn-tests/blob/c9e603cf6815030305d8a0e8e504251b9431243d/manager/integration/tests/test_settings.py#L785C1-L786C62.). 
     
It was noted that there are two possible scenarios:
Volume 2's rebuilding starts first, and volume 1's rebuilding is delayed until volume 2 becomes healthy.
Volume 1's rebuilding starts first, and volume 2's rebuilding waits until volume 1 becomes healthy.

The exact order of rebuilding initiation may vary, our enhancement ensures that both volumes have been successfully under rebuilding.

Reply:
Then, this verification does not achieve what we want to check. Probably you can juts check there is only one rebuilding, then then all rebuilt successfully eventually. This can make the test meaningful and complete. WDYT?

Reply:
@roger-ryao Maybe you can try to print out all the rebuild status throughout the whole test case process, so we can have a clear picture of the rebuilding orders, which rebuilding is canceled/restarted, or any behavior that is unexpected.